downstreamadapter/dispatcher/event_dispatcher.go

```diff
@@ -234,7 +149,54 @@ func (d *Dispatcher) InitializeTableSchemaStore(schemaInfo []*heartbeatpb.Schema
 	return true, nil
 }
 
-// HandleEvents can batch handle events about resolvedTs Event and DML Event.
+func (d *EventDispatcher) HandleCacheEvents() {
+	select {
+	case cacheEvents, ok := <-d.cacheEvents.events:
+		if !ok {
+			return
+		}
+		block := d.HandleEvents(cacheEvents.events, cacheEvents.wakeCallback)
+		if !block {
+			cacheEvents.wakeCallback()
+		}
+	default:
+	}
+}
+
+func (d *EventDispatcher) cache(dispatcherEvents []DispatcherEvent, wakeCallback func()) {
+	d.cacheEvents.Lock()
+	defer d.cacheEvents.Unlock()
+	if d.GetRemovingStatus() {
+		log.Warn("dispatcher has removed", zap.Any("id", d.id))
+		return
+	}
+	// cache here
+	cacheEvents := newCacheEvents(dispatcherEvents, wakeCallback)
+	select {
+	case d.cacheEvents.events <- cacheEvents:
+		log.Warn("cache event",
+			zap.Stringer("dispatcher", d.id),
+			zap.Uint64("dispatcherResolvedTs", d.GetResolvedTs()),
+			zap.Int("length", len(dispatcherEvents)),
+			zap.Int("eventType", dispatcherEvents[len(dispatcherEvents)-1].Event.GetType()),
+			zap.Uint64("commitTs", dispatcherEvents[len(dispatcherEvents)-1].Event.GetCommitTs()),
+			zap.Uint64("redoGlobalTs", d.redoGlobalTs.Load()),
+		)
+	default:
+		log.Panic("dispatcher cache events is full", zap.Stringer("dispatcher", d.id), zap.Int("len", len(d.cacheEvents.events)))
+	}
+}
+
+func (d *EventDispatcher) HandleEvents(dispatcherEvents []DispatcherEvent, wakeCallback func()) (block bool) {
+	// redo check
```

Comment:
Don't use short phrases as comments. Use clear sentences that explain what you did and what it does.

This is a general rule for all comments, please update all comments in this PR with this idea.